### PR TITLE
Improve test and release process

### PR DESCRIPTION
**Pull request by guilhermef**
It's the same build environment used in libthumbor and thumbor-aws.

This PR will add:
* Stalebot
* Dependabot
* CodeQL
* release workflow
* Linting on unittest workflow
* Unittest workflow dependency cache

### CODE REVIEW

**Comment by marcelometal** [comment]
Hi @guilhermef , 
Why [re-migrate](https://github.com/thumbor/remotecv/commit/1d71118d8e3ace7fb8306f65b2100cc1786bd075) to poetry? PEP 517?

**Comment by guilhermef** [comment]
It's mostly to be able to reuse the cached dependencies, and it makes it easier to pin dependencies.
The unittest workflow was also updated to benefit from it.

**Review by heynemann — APPROVED** [review]
If noone opposes, I'm good with this. Just don't do it in thumbor's main repo. Other than that, it should be fine.

**Comment by marcelometal** [comment]
IMHO, we can go without `poetry`, there is no real gain and we are adding another third-party dependency to the project

**Review by scorphus — COMMENTED** [review]
I also give `poetry` a strong 👎 We don't need it. And it's yet another dependency.

In fact, I think we should get rid of `poetry` entirely, thumbor-wise I mean. Yesterday I was debugging `libthumbor` and had to go over this loop:

- edit/save
- `poetry build`
- ...wait...
- `pip install dist/libthumbor-x.y.z.tar.gz`
- ...wait...
- experiment
- repeat

There's definitely no poetry in that, whatsoever. And, BTW, python-poetry/poetry#34 is a joke.

The other aspects of this PR are stellar.

**Comment by scorphus** [comment]
Also, https://github.com/thumbor/libthumbor/pull/52#discussion_r841120482... meh.

**Comment by guilhermef** [comment]
@scorphus you don't need to install the library.
Normally, I use `poetry run <bin>`

**Comment by guilhermef** [comment]
Poetry does offer GIthub actions to cache the build environment, it helps to speed up the build process and it pins the version across builds.

**Comment by guilhermef** [comment]
@scorphus, my suggestion would be to keep poetry while we don't have a better solution for environment caching on builds, and I'll add more docs to avoid confusion on how to run the projects with poetry.

**Comment by scorphus** [comment]
@guilhermef, what could other options be in your view?

> [...] Just don't do it in thumbor's main repo. Other than that, it should be fine.

@heynemann, why you think we shouldn't do this in thumbor's main repo?

**Comment by scorphus** [comment]
@marcelometal what are your thoughts? Apart from previous words...

Also, do we package remotecv? What's your experience with debian packages that use poetry?

**Comment by marcelometal** [comment]
Hi @scorphus , sorry for the late reply...

Would be nice packaging remoteCV for Debian =)

IMHO, we should avoid to include third-party libs, because it increase the complexity... 
The latest upstream version is not the same in Debian and should not have all the features we will use, for example.

[Debian: poetry](https://packages.debian.org/sid/python3-poetry-core): 1.0.7
[Upstream: poetry](https://github.com/python-poetry/poetry/releases): 1.1.13

**Comment by guilhermef** [comment]
Ok, then I'll remove poetry and the environment cache.

**Comment by guilhermef** [comment]
@scorphus @marcelometal I've removed Poetry dependency.

File: .github/codeql/codeql-config.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+name: "thumbor-aws CodeQL config"
```
Comment (scorphus) [inline .github/codeql/codeql-config.yml:1]:
```suggestion
name: "remotecv CodeQL config"
```

File: .github/workflows/unittest.yaml [hunk 2]
```diff
@@ -6,18 +7,51 @@ jobs:
...
         run: make setup
-      - run: make test
+      - name: Start Redis
+        run: make run-redis
+      - name: Run Unit Tests
+        run: make unit
+        env:
+          # Only used for satisfying boto
+          AWS_ACCESS_KEY_ID: foobar
+          AWS_SECRET_ACCESS_KEY: foobar
```
Comment (scorphus) [inline .github/workflows/unittest.yaml:37]:
```suggestion
```

File: flake8 [hunk 3]
```diff
@@ -1,6 +0,0 @@
-[flake8]
-ignore = W801,E501,W605,W504,W606,W503,E203,E266
-max-line-length = 88
-max-complexity = 20
-select = B,C,E,F,W,T4
-exclude = ./.tox/*,./build/*,./docs/conf.py
```
Comment (scorphus) [inline flake8:6]: 👍

File: setup.cfg [hunk 4]
```diff
@@ -4,10 +4,3 @@ include_trailing_comma=True
 force_grid_wrap=0
 use_parentheses=True
 line_length=88
-
-[flake8]
-ignore = W801,E501,W605,W504,W606,W503,E203,E266
-max-line-length = 88
-max-complexity = 20
-select = B,C,E,F,W,T4
-exclude = ./.tox/*,./build/*,./docs/conf.py
```
Comment (scorphus) [inline setup.cfg:13]: 👍 Good catches

**Review by scorphus — COMMENTED** [review]
LGTM 👍 Thanks, @guilhermef

Please consider a couple suggestions below.

**Pull request merged**